createHandler.web.js

```diff
@@ -0,0 +1,203 @@
+    };
+
+    hasBegun = false;
+    touchBank = [];
+    timeout = null;
+
+    setNativeProps() {}
+
+    clearState = () => {
+      this.hasBegun = false;
+      this.touchBank = [];
+      window.clearTimeout(this.timeout);
+    };
+
+    handlePressIn = event => {
+      const { maxDelayMs } = this.props;
+
+      if (!this.hasBegun) {
+        event.persist();
+        this.hasBegun = true;
+        handleBegan(this.props, event);
+
+        // Cancel if not finished in time
+        this.timeout = window.setTimeout(() => {
+          if (this.hasBegun) {
+            this.clearState();
+            handleFailed(this.props, event);
+          }
+        }, maxDelayMs);
+      }
+    };
+
+    handlePressOut = event => {
+      const {
+        touchHistory: { touchBank = [] },
+      } = event;
+      const { maxDeltaX, maxDeltaY, maxDurationMs, numberOfTaps } = this.props;
+
+      this.touchBank = this.touchBank.concat(touchBank);
+
+      // Check if all touches are valid
+      const areTouchesValid = this.touchBank.every(touch => {
+        const {
+          currentPageX,
+          currentPageY,
+          currentTimeStamp,
+          startPageX,
+          startPageY,
+          startTimeStamp,
+        } = touch;
+
+        // Check if touch took longer than it may
+        if (startTimeStamp + maxDurationMs < currentTimeStamp) {
+          return false;
+        }
+
+        // Check if touch moved too far away
+        if (
+          startPageX + maxDeltaX < currentPageX ||
+          startPageX - maxDeltaX > currentPageX
+        ) {
+          return false;
+        }
+
+        // Check if touch moved too far away
+        if (
+          startPageY + maxDeltaY < currentPageY ||
+          startPageY - maxDeltaY > currentPageY
+        ) {
+          return false;
+        }
+
+        return true;
+      });
+
+      // Check if all touches were valid and the necessary number of touches was achieved
+      if (!areTouchesValid) {
+        this.clearState();
+        handleFailed(this.props, event);
+      } else if (this.touchBank.length >= numberOfTaps) {
+        handleActivate(this.props, event);
+        this.clearState();
+        handleEnd(this.props, event);
+      }
+    };
+
+    render() {
+      const { children, style } = this.props;
+
+      return (
+        <TouchableWithoutFeedback
+          style={style}
+          onPressIn={this.handlePressIn}
+          onPressOut={this.handlePressOut}>
+          {children}
+        </TouchableWithoutFeedback>
+      );
+    }
+  },
+};
+
+export default function createHandler(handlerName, propTypes = {}) {
+  class Handler extends React.Component {
+    static displayName = handlerName;
+
+    static propTypes = propTypes;
+
+    _refHandler = node => {
+      this._viewNode = node;
+    };
+
+    setNativeProps = (...args) => {
+      this._viewNode.setNativeProps(...args);
+    };
+
+    render() {
+      let Handler = handlers[handlerName];
```

Comment:
nit: I think this check could be moved outside of the render method and directly to `createHandler`